### PR TITLE
Grid v4.x.x

### PR DESCRIPTION
**Pull request by scamden**
@joechrz @joemunriq This is a pull request to discuss v4 of the grid. I'll probably also create a milestone so we can create issues and mark them v4

The first thing i did was move the angular adapter code over to grid-angular-adapter. (The test app still uses angular to setup but i'll rip that out soon.)

Next steps: I'll probably port the build over to webpack and setup typescript. I'll create an issue so we can discuss that if you have any thoughts. I'll likely leave the tests running in karma/jasmine/phantom since phantom 2 is pretty fast and I want to make sure the tests run as originally intended during all these refactors.

Def let me know any thoughts you have about things we want to include in the v4 release, especially things that require breaking changes. I'm trying to get the typescript conversion done relatively quickly so ideally we can get any necessary breaking changes done quickly first and then we can lump on features as minor releases after.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,67 +14,80 @@
...
-    "aliasify": "^1.5.0",
-    "browserify": "~12.0.1",
-    "browserify-istanbul": "~0.2.1",
-    "chokidar-cli": "~1.1.1",
+    "autoprefixer": "~6.4.0",
+    "awesome-typescript-loader": "~3.1.3",
+    "babel-cli": "~6.24.1",
+    "babel-loader": "~6.2.5",
+    "babel-preset-es2015": "~6.18.0",
+    "babel-preset-stage-1": "~6.13.0",
```
Comment (joechrz) [inline package.json]: what's babel used for?

**Comment by joechrz** [comment]
<3 awesome first step.  really hope i have a good use case for this again someday

File: package.json (hunk 1 above)
Comment (scamden) [inline package.json]: originally had it to allow es6 in webpack config, but that didn't play nice with karma so it may be unused now

Comment (joechrz) [inline package.json]: 👍 cool- didn't see anything that needed it so was curious.  even for the webpack config, that's a pretty huge dep to add for a single file that doesn't change too often :)

Comment (joechrz) [inline package.json]: wait. webpack config does have 1 arrow function.

**Comment by scamden** [comment]
oh so there is.. i guess i am still running it though babel in dev mode,
just cause that's how we have it in creditiq and a copied the setup more or
less. really should be necessary

**Comment by scamden** [comment]
i have released this under the "next" dist-tag after testing will merge to master for public release

**Pull request merged**